client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -98,7 +119,7 @@ function useAddRenewalItems( { originalPurchaseId, productAlias, setState } ) {
 					selectedSiteSlug
 				);
 			} )
-			.filter( Boolean );
+			.filter( < T >( x: T ): x is Exclude< T, null > => !! x );
```

Comment:
This is nice. It's almost worth turning this into a common utility function.

Reply:
Yeah, that took me a while to figure out after reading a lot of links on the web. I can't recall where I found this one exactly but extracting it to its own helper would be really helpful.

Reply:
Filtering out "empty" items is a pretty common practice. You could extend this even further to exclude any "falsy" value in the type guard.